### ISSUE TITLE
Rule idea: Extract repetitive casts into a variable

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#9488]
Similar to S3247, we could report on what #9482 tried to implement (and didn't work within S3247 implementation).

It should be a new rule, focused only on direct repetitive casts `(Fruit)x`

The rule should report when there are 3 or more repetitions, as two are not worth the extraction.

Ideally, it would report also for nested blocks. And stopped counting on variable assignments.

Original test cases:
```
    public void MultipleCasts_RootBlock(object arg)
    {
        _ = (Fruit)arg; // Noncompliant {{Extract this repetitive cast into a variable.}}
        //  ^^^^^^^^^^
        _ = (Fruit)arg;             // Secondary
        _ = (Fruit/*Comment*/)arg;  // Secondary
        _ = (  Fruit   )arg;        // Secondary with extra whitespace
    }

    public void MultipleCasts_SameBlock(object arg)
    {
        if (true)
        {
            _ = (Fruit)arg; // Noncompliant {{Extract this repetitive cast into a variable.}}
            _ = (Fruit)arg; // Secondary
            _ = (Fruit)arg; // Secondary
        }
    }

    public void MultipleCasts_NestedBlock(object arg)
    {
        _ = (Fruit)arg;         // Noncompliant
        if (true)
        {
            _ = (Fruit)arg;     // Secondary
            foreach (var ch in "Lorem ipsum")
            {
                _ = (Fruit)arg; // Secondary
                _ = (Fruit)arg; // Secondary
            }
            _ = (Fruit)arg;     // Secondary
        }
    }

    public void MultipleCasts_Lambda(object arg)
    {
        _ = (Fruit)arg; // Noncompliant
        Action a = () =>
        {
            _ = (Fruit)arg; // Secondary
            _ = (Fruit)arg; // Secondary   
        };
        Func<object, int> f = x =>
        {
            _ = (Fruit)arg;  // Secondary
            _ = (Fruit)x;
            return 0;
        };
    }

    public void MultipleCastsDifferentBlocks(object arg)
    {
        if (true)
        {
            _ = (Fruit)arg; // Compliant, we only look into the current and nested blocks
        }

        while(false)
        {
            _ = (Fruit)arg;
        }

        _ = (Fruit)arg;

        switch (42)
        {
            case 1:
                _ = (Fruit)arg; // Compliant, not the same block
                break;
            case 2:
                _ = (Fruit)arg;
                break;
        }
    }
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1576](https://sonarsource.atlassian.net/browse/NET-1576)

[NET-1576]: https://sonarsource.atlassian.net/browse/NET-1576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
